shared/actions/profile.js

```diff
@@ -88,47 +96,174 @@ function updateUsername (username: string): UpdateUsername {
   }
 }
 
-function cancelAddProof () : AsyncAction {
+function cancelAddProof (): AsyncAction {
   return (dispatch) => {
-    if (submitUsernameResponse) {
-      engine.cancelRPC(submitUsernameResponse, InputCancelError)
-      submitUsernameResponse = null
+    dispatch(updateError(null))
+    if (promptUsernameResponse) {
+      engine.cancelRPC(promptUsernameResponse, InputCancelError)
+      promptUsernameResponse = null
+    }
+
+    if (outputInstructionsResponse) {
+      engine.cancelRPC(outputInstructionsResponse, InputCancelError)
+      outputInstructionsResponse = null
     }
+
     dispatch(navigateUp())
   }
 }
 
-function addProof (platform: PlatformsExpanded): AsyncAction {
+function updateProofText (proof: string): UpdateProofText {
+  return {
+    type: Constants.updateProofText,
+    payload: {proof},
+  }
+}
+
+function updateError (error: ?string, code: ?number): UpdateError {
+  // Flow needs this for some reason instead of default params
+  if (code === undefined || code === null) {
```

Comment:
🐑 You can do `code == null` and that covers both cases, although that's implied, so this might be preferred to be more explicit